src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,336 @@
+        if (details.equals("")) {
+            this.isValidDetail = false;
+        } else {
+            this.isValidDetail = true;
+        }
+        this.details = details;
+    }
+
+    /**
+     * Executes the function based on the command and details provided.
+     *
+     * @param tasks The task list for addition of task.
+     * @param storage The storage object for reading or writing tasks into a specific file.
+     * @return Task object based on the input command.
+     * @throws NoSuchCommandException When an invalid command is inputted.
+     * @throws InvalidIndexException If an invalid task index is inputted.
+     * @throws EmptyDescriptionException If a task description is not inputted.
+     * @throws UnmatchedArgumentException If unmatched arguments are inputted.
+     */
+    public String execute(TaskList tasks, Storage storage) throws NoSuchCommandException, InvalidIndexException,
+            UnmatchedArgumentException, EmptyDescriptionException {
+
+        String result = "";
+        if (command.equals("list")) {
+            result = tasks.listAllTask();
+        } else if (command.equals("mark")) {
+
+            if (!isValidDetail) {
+                throw new InvalidIndexException();
+            }
+            int index = Integer.parseInt(details);
+            if (index > 0 && index <= tasks.size()) {
+                result = mark(index, tasks, storage);
+            } else {
+                throw new InvalidIndexException();
+            }
+        } else if (command.equals("delete")) {
+
+            if (!isValidDetail) {
+                throw new InvalidIndexException();
+            }
+            int index = Integer.parseInt(details);
+            if (index > 0 && index <= tasks.size()) {
+                result = delete(index, tasks, storage);
+            } else {
+                throw new InvalidIndexException();
+            }
+        } else if (command.equals("unmark")) {
+
+            if (!isValidDetail) {
+                throw new InvalidIndexException();
+            }
+            int index = Integer.parseInt(details);
+            if (index > 0 && index <= tasks.size()) {
+                result = unmark(index, tasks, storage);
+            } else {
+                throw new InvalidIndexException();
+            }
+        } else if (command.equals("todo")) {
+            result = addTodo(details, tasks, storage);
+        } else if (command.equals("deadline")) {
+            result = addDeadline(details, tasks, storage);
+        } else if (command.equals("event")) {
+            result = addEvent(details, tasks, storage);
+        } else if (command.equals("due")) {
+            result = checkTaskDue(details, tasks);
+        } else if (command.equals("find")) {
+            result = find(details, tasks);
+        } else if (command.equals("bye")) {
+            isExit();
+        } else {
+            throw new NoSuchCommandException();
+        }
+        return result;
+    }
```

Comment:
I can understand the logical flow behind the block, but is there anyway we can make it more readable? Maybe extract out eg.mark to check for valid details and parsing instead of doing everything here?